validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -104,10 +108,23 @@ public void onSlot(UInt64 slot) {
       UInt64 analyzedEpoch = currentEpoch.minus(ATTESTATION_INCLUSION_RANGE);
       AttestationPerformance attestationPerformance =
           getAttestationPerformanceForEpoch(currentEpoch, analyzedEpoch);
-      statusLogger.performance(attestationPerformance.toString());
+      switch (mode) {
+        case LOGGING:
+          statusLogger.performance(attestationPerformance.toString());
+          break;
+        case METRICS:
+          validatorPerformanceMetrics.updateAttestationPerformanceMetrics(attestationPerformance);
+          break;
+        case ALL:
+          statusLogger.performance(attestationPerformance.toString());
+          validatorPerformanceMetrics.updateAttestationPerformanceMetrics(attestationPerformance);
+          break;
+        case NONE:
+          throw new IllegalStateException(
+              "Performance Tracker should not be running in NONE mode.");
```

Comment:
nit: Possibly simpler as:
```suggestion
      if (mode.isLoggingEnabled()) {
          statusLogger.performance(attestationPerformance.toString());
     }
     if (mode.isMetricsEnabled()) {
        validatorPerformanceMetrics.updateAttestationPerformanceMetrics(attestationPerformance);
     }
```

Reply:
Done.